core/src/main/java/io/methvin/watcher/DirectoryWatcher.java

```diff
@@ -139,6 +139,15 @@ public void watch() {
             if (Files.isDirectory(childPath, NOFOLLOW_LINKS)) {
               if (!Boolean.TRUE.equals(fileTreeSupported)) {
                 registerAll(childPath);
+              } else {
```

Comment:
This `else` block will happen if either `FILE_TREE` is supported or if we haven't tested yet if it's supported (which `registerAll` does). I don't think that's what we want.

I actually think we should walk the tree no matter what the support of `FILE_TREE` is, since when we have to manually register the listener, it is possible to miss events before we start listening. Perhaps this could be `if (!isMac)`, since our custom mac watcher is the only implementation we can be relatively confident doesn't lose events.

Finally, we should be using the same hashing logic as below to check if we've already seen the file. That will help prevent duplicate events. Maybe that should be extracted into a helper method.